src/Processors/ConcatProcessor.cpp

```diff
@@ -4,6 +4,17 @@
 namespace DB
 {
 
+ConcatProcessor::ConcatProcessor(const Block & header, size_t num_inputs)
+    : IProcessor(InputPorts(num_inputs, header), OutputPorts{header}), current_input(inputs.begin())
+{
+}
+
+void ConcatProcessor::prepareInitializeInputs()
```

Comment:
I'm afraid this brakes pretty important invariant of `ConcatProcessor`: we don't activate any input until it's turn to read from. I will improve comment.

Reply:
But, generally, you right. Let's just replace `ConcatProcessor` to `Resize`.